tests/common_subset.rs

```diff
@@ -31,21 +30,23 @@ fn test_common_subset<A: Adversary<CommonSubset<NodeUid>>>(
     }
 
     // Terminate when all good nodes do.
-    while network
-        .nodes
-        .values()
-        .any(|node| network.adv_nodes.contains(&node.algo.our_id()) || node.algo.terminated())
-    {
-        let id = network.step();
-        if let Some(output) = network.nodes[&id].outputs().iter().next() {
-            assert_eq!(inputs, output);
-            debug!("Node {:?} decided: {:?}", id, output);
-
-            // Test uniqueness of output of the good nodes.
-            if !network.adv_nodes.contains(&id) {
-                assert!(!decided_nodes.insert(id));
-            }
+    while !network.nodes.values().all(TestNode::terminated) {
+        network.step();
+    }
+    // Verify that all instances output the same set.
+    let mut expected = None;
+    for node in network.nodes.values() {
+        if let Some(output) = expected.as_ref() {
+            assert!(iter::once(output).eq(node.outputs()));
+            continue;
         }
+        assert_eq!(1, node.outputs().len());
+        expected = Some(node.outputs()[0].clone());
+    }
+    let output = expected.unwrap();
+    assert!(output.len() * 3 > inputs.len() * 2);
```

Comment:
Can you please make a comment on what condition this assertion is checking?

Reply:
If my understanding is correct, the Common Subset algorithm guarantees that the output contains more than ⅔ of the values, so we should assert that.